stdlib/public/core/Reverse.swift

```diff
@@ -12,34 +12,47 @@
 
 /// An index that traverses the same positions as an underlying index,
 /// with inverted traversal direction.
-public struct ReverseIndex<Base> : Comparable {
-  public init(_ base: Base) { self.base = base }
+public struct ReverseIndex<Base : BidirectionalCollection> : Comparable {
```

Comment:
I was attempting to constrain things to match collection requirements for reverse and random access.

Are you saying instead of...
```
public struct ReverseIndex<Base : BidirectionalCollection> : Comparable {
  public let base: Base.Index
}
public typealias Index = ReverseIndex<Base>
```
...we should...
```
public struct ReverseIndex<Base> : Comparable {
  public let base: Base
}
public typealias Index = ReverseIndex<Base.Index>
```